### PR TITLE
fixing addFile while original file size changes during archiving

### PR DESCRIPTION
**Pull request by ovidiul**
When adding a file which size is being changed during archiving, the archive would become corrupt because the size in the header is different than the size of the actual data written. So an additional check should be done and if the sizes are different, the header should be updated. A quick test would be to create an infinite loop where you write 1 byte to a test.txt file and then trying to add that file to the archive and extracting the archive. 

The error one would get on extraction is "Header does not match it's checksum" because the next file added would not be fetched correctly, and it's header is not seeked at the correct point.

### CODE REVIEW

File: src/Tar.php [hunk 1]
```diff
@@ -248,10 +248,11 @@ public function addFile($file, $fileinfo = '')
         if (!$fp) {
             throw new ArchiveIOException('Could not open file for reading: '.$file);
         }
-
+		
         // create file header
+        $archive_header_position = ftell($this->fh);
         $this->writeFileHeader($fileinfo);
-
+
```
Comment (splitbrain) [inline src/Tar.php]: introduced trailing whitespace

File: src/Tar.php [hunk 2]
```diff
@@ -264,6 +265,22 @@ public function addFile($file, $fileinfo = '')
...
+        {
+			$archive_current_position = ftell($this->fh);
+			fseek($this->fh, $archive_header_position);
+			
+			$fileinfo->setSize(ftell($fp));
+			$this->writeFileHeader($fileinfo);
+
+			fseek($this->fh, $archive_current_position);
+
+		}
```
Comment (splitbrain) [inline src/Tar.php]: broken indentation

**Review by splitbrain — CHANGES_REQUESTED** [review]
Could you please fix the formatting errors, check why the existing test cases break and add a specific test for the actual problem here?

**Comment by ovidiul** [comment]
Can't seem to pass the HHVM check, it fails on ftell function saying tell is not supported, do you have any experience maybe with this and a way to overcome it?

**Comment by splitbrain** [comment]
It's weird. From what I can see hhvm should support ftell. It might be that travis is running a rather outdated release. However I was thinking about the approach to rewrite the header size when a file size change was detected.

All we know is that the file changed between when we started reading it and when we finished. We might have read garbage and just changing the header will not fix that. I think the cleaner approach would be to throw an exception and let the user handle it.

**Comment by splitbrain** [comment]
Okay I figured out what's going on. You initially had problems with $this->fh not being a resource. This happens when a tbz is written (the file is openend with bzopen). There is no tell or seek support in bz streams which is what hhvm complains about.

I don't see a way to implement the feature as you imagined it with supporting bz2. The only way would be reading the whole file into memory and check it's length before writing it to the archive.

I'll add an exception.

**Pull request closed without merging**